File: demo/testAnimation/ViewController.m
//
//  ViewController.m
//  testAnimation
//
//  Created by Vlad Kosyi on 5/6/16.
//  Copyright © 2016 Vlad Kosyi. All rights reserved.
//

#import "ViewController.h"
#import "GifButton.h"
#import "UIImage+animatedGIF.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet GifButton *gifButton;
@property (strong, nonatomic) NSArray *imageArray;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
//    NSMutableArray *tempImageArray = [[NSMutableArray alloc]init];
//    
//    for (int i = 0; i < 39; i++) {
//        [tempImageArray addObject:[UIImage imageNamed:[NSString stringWithFormat:@"%d",i]]];
//    }
//  
//  self.imageArray = [NSArray arrayWithArray:tempImageArray];
  

//  NSURL *url = [[NSBundle mainBundle] URLForResource:@"gif" withExtension:@"gif"];
//  self.gifButton.imageView.image = [UIImage animatedImageWithAnimatedGIFData:[NSData dataWithContentsOfURL:url]];

//  self.imageView.image = [UIImage animatedImageWithAnimatedGIFData:[NSData dataWithContentsOfURL:url]];
//  self.imageArray = [UIImage animatedImageArrayWithAnimatedGIFData:[NSData dataWithContentsOfURL:url]];
//  self.gifButton.imagesArray = self.imageArray;
//  self.gifButton.animationDuration = 2;
//  self.gifButton.repeatCount = 1;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];

}
- (IBAction)gifImageAction:(id)sender {

    [self.gifButton startAnimation];
  

}


@end

File: demo/KVGifButtonSwift/ViewController.swift
//
//  ViewController.swift
//  KVGifButtonSwift
//
//  Created by Vlad Kosyi on 5/10/16.
//  Copyright © 2016 Vlad Kosyi. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var imageArray: [UIImage] = []
    @IBOutlet weak var gifButton: GifButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        imageArray = [UIImage]()
      
      for i in 0 ..< 39 {
        self.imageArray.append(UIImage(named: "\(i)")!)
      }

        gifButton.imagesArray.appendContentsOf(imageArray)
        gifButton.repeatCount = 1
        gifButton.animationDuration = 2
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    @IBAction func gifImageAction(sender: AnyObject) {
        gifButton.startAnimation()
    }

}

